addons/base_vat/models/res_partner.py

```diff
@@ -168,27 +175,30 @@ def check_vat(self):
                 msg = partner._build_vat_error_message(country and country.code.lower() or None, partner.vat, partner_label)
                 raise ValidationError(msg)
 
-    @api.onchange('vat', 'country_id')
-    def _onchange_check_vies(self):
-        """ Check the VAT number with VIES, if enabled. Return a non-blocking warning if the check fails."""
-        if self.env.context.get('company_id'):
-            company = self.env['res.company'].browse(self.env.context['company_id'])
-        else:
-            company = self.env.company
-        if not company.vat_check_vies:
+    @api.depends('vies_vat_to_check')
+    def _compute_vies_valid(self):
+        """ Check the VAT number with VIES, if enabled."""
+        if not self.env['ir.config_parameter'].get_param('base_vat.perform_vies_check'):
+            self.write({'vies_valid': False})
             return
 
-        eu_countries = self.env.ref('base.europe').country_ids
-        for eu_partner_company in self.filtered(lambda partner: partner.country_id in eu_countries and partner.is_company):
-            # Skip checks when only one character is used. Some users like to put '/' or other as VAT to differentiate between
-            # A partner for which they didn't input VAT, and the one not subject to VAT
-            if not eu_partner_company.vat or len(eu_partner_company.vat) == 1:
+        for partner in self:
+            if not partner.vies_vat_to_check:
+                partner.vies_valid = False
                 continue
-            country = eu_partner_company.country_id
-            if self._run_vies_test(eu_partner_company.vat, country) is False:
-                self.vies_failed_message = _("The VAT number %s failed the VIES VAT validation check.", eu_partner_company.vat)
-            else:
-                self.vies_failed_message = False
+            try:
+                vies_valid = check_vies(partner.vies_vat_to_check, timeout=10)
+                partner.vies_valid = vies_valid['valid']
+            except Exception as e:
+                if partner._origin.id:
+                    msg = ""
+                    if isinstance(e, OSError):
+                        msg = _("Connection with the VIES server failed. The VAT number %s could not be validated.", partner.vies_vat_to_check)
+                    elif isinstance(e, InvalidComponent):
+                        msg = _("The VAT number %s could not be interpreted by the VIES server.", partner.vies_vat_to_check)
+                    partner._origin.message_post(body=msg)
+                _logger.exception("The VAT number %s failed VIES check with error: %s.", partner.vies_vat_to_check, str(e))
```

Comment:
`logging.exception` will already include the complete traceback in the logs, no need to repeat it
```suggestion
                _logger.exception("The VAT number %s failed VIES check.", partner.vies_vat_to_check)
```